Library/Source/NativeEngine.cpp

```diff
@@ -151,6 +151,59 @@ namespace babylon
             }
         }
 
+        void GenerateMipmap(const uint8_t* const __restrict source, const int size, const int channels, uint8_t* __restrict dest)
```

Comment:
Is `__restrict` a thing we can have here? I'd rather have these buffers be gsl::spans backed by vectors, just for the safety and clarity, as opposed to raw pointers; but that's more of a nit.

Reply:
It's definitely something useful here :) The performance difference is huge, especially with this case.
Do using  gsl::span prevent memory aliasing with uint8_t ?

Reply:
Is the performance huge from the __restrict or raw pointers?

Reply:
The main issue with `__restrict` (now that I've read more on it), is that it's not part of C++ standard. Different compilers use different symbols. If we really want this, it needs to be part of a macro definition.

gsl::span appears to add range checks to each access which would also be bad for performance here.